src/frontend/src/flows/recovery/displaySeedPhrase.ts

```diff
@@ -1,9 +1,52 @@
-import { html, render } from "lit-html";
+import { html } from "lit-html";
+import { withRef, renderPage } from "../../utils/lit-html";
+import { ref, createRef, Ref } from "lit-html/directives/ref.js";
 import { checkmarkIcon, copyIcon } from "../../components/icons";
 import { mainWindow } from "../../components/mainWindow";
 import { toast } from "../../components/toast";
 
-const pageContent = (seedPhrase: string) => {
+const displaySeedPhraseTemplate = ({
+  seedPhrase,
+  onContinue,
+  copyPhrase: copyPhrase_,
+}: {
+  seedPhrase: string;
+  copyPhrase: () => Promise<void>;
+  onContinue: () => void;
+}) => {
+  const phraseElement: Ref<HTMLElement> = createRef();
+  const phraseCopyElement: Ref<HTMLElement> = createRef();
+
+  const continueButton: Ref<HTMLButtonElement> = createRef();
+  const checkbox: Ref<HTMLInputElement> = createRef();
+
+  // Selects the phrase content to give visual feedback of what's been copied
+  const selectPhrase = () => {
+    withRef(phraseElement, (phraseElement) => {
+      const selection = window.getSelection();
+      if (selection !== null) {
+        const range = document.createRange();
+        range.selectNodeContents(phraseElement);
+        selection.removeAllRanges();
+        selection.addRange(range);
```

Comment:
I generally detach the range afterwards to free memory

```
...
selection.addRange(range);

range.detach();
```

Reply:
Thanks! I've only moved the code for now; will do this subsequently with the other fix you suggested above.